security/tinc/src/opnsense/mvc/app/models/OPNsense/Tinc/Tinc.xml

```diff
@@ -37,10 +37,18 @@
                     <WildcardEnabled>N</WildcardEnabled>
                 </intaddress>
                 <subnet type="NetworkField">
-                    <Required>Y</Required>
+                    <Required>N</Required>
                     <WildcardEnabled>N</WildcardEnabled>
                     <NetMaskRequired>Y</NetMaskRequired>
                     <FieldSeparator>,</FieldSeparator>
+                    <Constraints>
+                        <check001>
+                            <ValidationMessage>This field must be set in router mode.</ValidationMessage>
+                            <type>SetIfConstraint</type>
+                            <field>mode</field>
+                            <check>router</check>
```

Comment:
The field `mode`  should have a reference to enforce validation when it changes (https://docs.opnsense.org/development/frontend/models_constraints.html?highlight=setifconstraint#dependconstraint)

Reply:
Ok, I implemented the reference constraint, see https://github.com/opnsense/plugins/pull/1733/commits/e19f92d1948f737bc005066aecf0c8176ddf2cc9, and it virtually works. However, when you change switch mode to router mode in the WebUI and your subnet is empty, the error ("... field must be set ...") is displayed next to router field while it should naturally be displayed next to subnet field as it is subnet, not mode, that fails validation.

Reply:
I have to check that out locally, at a first glance I would expect it to show on the subnet field too.

Reply:
@vnxme can you try https://github.com/opnsense/core/commit/2eb1ee5f53ec4cc627ab017766dcdf5591a056c0 ? this should fix this specific issue.

Reply:
I confirm the fix you suggest highlights the correct field (empty host.subnet) when host.mode field changes value from switch to router. Thank you!

Reply:
@vnxme thanks for testing!